### ISSUE TITLE
Model with entity sets before entity types not parsing

### ISSUE THREAD

**Issue by Involture** [odata2ts/odata2ts#132]
Some models contains two schema in their `DataServices` tag : one containing the entity container and the entity sets and the other one containing the entity types.
One such model is [Northwind V2](https://services.odata.org/v2/northwind/northwind.svc/$metadata).

Moreover, some models (for example some from the sap world) have the entity set schema before the entity types one.

To illustrate this, one can find below the metadata file of Northwind v2 with the two schema swapped.
[northwindv2metadataReordered.zip](https://github.com/odata2ts/odata2ts/files/10726900/northwindv2metadataReordered.zip)

The resulting file structure is the following
```
<?xml version="1.0" encoding="UTF-8" standalone="yes"?>
<edmx:Edmx Version="1.0" xmlns:edmx="http://schemas.microsoft.com/ado/2007/06/edmx" class="TridactylThemeDark">
  <edmx:DataServices xmlns:m="http://schemas.microsoft.com/ado/2007/08/dataservices/metadata" m:DataServiceVersion="1.0">
    <Schema Namespace="ODataWeb.Northwind.Model" xmlns:d="http://schemas.microsoft.com/ado/2007/08/dataservices" xmlns:m="http://schemas.microsoft.com/ado/2007/08/dataservices/metadata" xmlns="http://schemas.microsoft.com/ado/2008/09/edm">
      <EntityContainer Name="NorthwindEntities" p7:LazyLoadingEnabled="true" m:IsDefaultEntityContainer="true" xmlns:p7="http://schemas.microsoft.com/ado/2009/02/edm/annotation">
        <EntitySet Name="Categories" EntityType="NorthwindModel.Category"/>
           ...
      </EntityContainer>
    </Schema>
    <Schema Namespace="NorthwindModel" xmlns:d="http://schemas.microsoft.com/ado/2007/08/dataservices" xmlns:m="http://schemas.microsoft.com/ado/2007/08/dataservices/metadata" xmlns="http://schemas.microsoft.com/ado/2008/09/edm">
      <EntityType Name="Category">
        <Key>
          <PropertyRef Name="CategoryID"/>
        </Key>
        <Property Name="CategoryID" Type="Edm.Int32" Nullable="false" p8:StoreGeneratedPattern="Identity" xmlns:p8="http://schemas.microsoft.com/ado/2009/02/edm/annotation"/>
        <Property Name="CategoryName" Type="Edm.String" Nullable="false" MaxLength="15" Unicode="true" FixedLength="false"/>
        <Property Name="Description" Type="Edm.String" Nullable="true" MaxLength="Max" Unicode="true" FixedLength="false"/>
        <Property Name="Picture" Type="Edm.Binary" Nullable="true" MaxLength="Max" FixedLength="false"/>
        <NavigationProperty Name="Products" Relationship="NorthwindModel.FK_Products_Categories" FromRole="Categories" ToRole="Products"/>
      </EntityType>
      ...
    </Schema>
  </edmx:DataServices><iframe xmlns="http://www.w3.org/1999/xhtml" class="cleanslate hidden" src="$metadata_fichiers/commandline.htm" id="cmdline_iframe" loading="lazy" style="height: 0px !important;"></iframe>
</edmx:Edmx>
```

When trying to generate artifactsodata2ts fails with the following error : 
```
Starting generation process
Reading file: $metadata.xml
Error while running the program Error: Association end couldn't be determined for NavigationProperty [Products]
    at C:\Users\Marti\Documents\Odata2ts\node_modules\@odata2ts\odata2ts\src\data-model\DataModelDigestionV2.ts:44:17
    at Array.map (<anonymous>)
    at DigesterV3.getNavigationProps (C:\Users\Marti\Documents\Odata2ts\node_modules\@odata2ts\odata2ts\src\data-model\DataModelDigestionV2.ts:39:36)
    at DigesterV3.getBaseModel (C:\Users\Marti\Documents\Odata2ts\node_modules\@odata2ts\odata2ts\src\data-model\DataModelDigestion.ts:98:55)        
    at DigesterV3.addEntityType (C:\Users\Marti\Documents\Odata2ts\node_modules\@odata2ts\odata2ts\src\data-model\DataModelDigestion.ts:134:30)      
    at DigesterV3.digestSchema (C:\Users\Marti\Documents\Odata2ts\node_modules\@odata2ts\odata2ts\src\data-model\DataModelDigestion.ts:82:10)        
    at DigesterV3.<anonymous> (C:\Users\Marti\Documents\Odata2ts\node_modules\@odata2ts\odata2ts\src\data-model\DataModelDigestion.ts:51:10)
    at Generator.next (<anonymous>)
    at C:\Users\Marti\Documents\Odata2ts\node_modules\tslib\tslib.js:167:75
    at new Promise (<anonymous>)
```

**Comment by texttechne** [comment]
Released with `@odata2ts/odata2ts@0.19.0`.
The generation for SuccessFactors works now in my tests. Please confirm and close.

**Comment by Involture** [comment]
Works like a charm
